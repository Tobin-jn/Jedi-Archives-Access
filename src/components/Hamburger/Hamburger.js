import React, {Component} from 'react';

import './Hamburger.css';

class Hamburger extends Component {
  constructor() {
    super()

    this.state = {
      topToggled: '',
      middleToggled: '',
      bottomToggled: '',
      topBladeToggled: '',
      middleBladeToggled: '',
      bottomBladeToggled: '',
      gripToggled: ''
    }
  }

  handleHamburger = () => {
    if (this.state.topToggled === '') {
      console.log('poo');
      this.setState({
        topToggled: 'top-toggled',
        middleToggled: 'middle-toggled',
        bottomToggled: 'bottom-toggled',
        topBladeToggled: 'top-blade-toggled',
        middleBladeToggled: 'middle-blade-toggled',
        bottomBladeToggled: 'bottom-blade-toggled',
        gripToggled: 'grip-toggled'
      })
    } else {
      this.setState({
        topToggled: '',
        middleToggled: '',
        bottomToggled: '',
        topBladeToggled: '',
        middleBladeToggled: '',
        bottomBladeToggled: '',
        gripToggled: ''
      }) 
    }
  }

  render() {
    return(
      <section 
        className='lightsaburger'
        onClick={() => {
          this.handleHamburger()
        }}
      >
        <div className={`top ${this.state.topToggled}`}>
         <div className={`top-blade blade ${this.state.topBladeToggled}`}></div>
         <div className={`top-grip grip ${this.state.gripToggled}`}></div>
        </div>
        <div className={`middle ${this.state.middleToggled}`}>
          <div className={`middle-grip grip ${this.state.gripToggled}`}></div>
          <div className={`middle-blade blade ${this.state.middleBladeToggled}`}></div>
        </div>
        <div className={`bottom ${this.state.bottomToggled}`}>
         <div className={`bottom-blade blade ${this.state.bottomBladeToggled}`}></div>
         <div className={`bottom-grip grip ${this.state.gripToggled}`}></div>
        </div>
      </section>
    )
  }
}

export default Hamburger