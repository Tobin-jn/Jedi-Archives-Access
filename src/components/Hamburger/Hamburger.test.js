import React from 'react'
import { shallow } from 'enzyme';

import Hamburger from './Hamburger'

describe('Hamburger', () => {
  let wrapper;
  let mockHamburgerChange = jest.fn()
  let mockLoginWarning = jest.fn()

  const deployedState = {
    topToggled: 'top-toggled',
    middleToggled: 'middle-toggled',
    bottomToggled: 'bottom-toggled',
    topBladeToggled: 'top-blade-toggled',
    middleBladeToggled: 'middle-blade-toggled',
    bottomBladeToggled: 'bottom-blade-toggled',
    gripToggled: 'grip-toggled',
    login: 'hide-login'
  }

  const undeployedState = {
    topToggled: '',
    middleToggled: '',
    bottomToggled: '',
    topBladeToggled: '',
    middleBladeToggled: '',
    bottomBladeToggled: '',
    gripToggled: '',
    login: 'hide-login'
  }

  beforeEach(() => {
    wrapper = shallow(<Hamburger 
      hamburgerChange={mockHamburgerChange}
      loginWarning={mockLoginWarning}
      ready={true}
    />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleHamburger on click',() => {
    let mockHandleHamburger = jest.fn()
    wrapper.instance().handleHamburger = mockHandleHamburger
    wrapper.find('.lightsaburger').simulate('click');

    expect(mockHandleHamburger).toHaveBeenCalled();
  });

  it('should update state on handleHamburger call if ready and unToggled', () => {
    wrapper.instance().handleHamburger();

    expect(wrapper.state()).toEqual(deployedState)
  });

  it('should update state on handleHamburger call if ready and toggled', async () => {
    await wrapper.setState({
      topToggled: 'top-toggled',
      middleToggled: 'middle-toggled',
      bottomToggled: 'bottom-toggled',
      topBladeToggled: 'top-blade-toggled',
      middleBladeToggled: 'middle-blade-toggled',
      bottomBladeToggled: 'bottom-blade-toggled',
      gripToggled: 'grip-toggled',
      login: 'hide-login'
    })
    wrapper.instance().handleHamburger();

    expect(wrapper.state()).toEqual(undeployedState)
  });

  it('should call hamburgerChange if ready and unToggled', () => {
    wrapper.instance().handleHamburger();

    expect(mockHamburgerChange).toHaveBeenCalled();
  });

  it('should call hamburgerChange if ready and toggled', async () => {
    await wrapper.setState({
      topToggled: 'top-toggled',
      middleToggled: 'middle-toggled',
      bottomToggled: 'bottom-toggled',
      topBladeToggled: 'top-blade-toggled',
      middleBladeToggled: 'middle-blade-toggled',
      bottomBladeToggled: 'bottom-blade-toggled',
      gripToggled: 'grip-toggled',
      login: 'hide-login'
    })

    wrapper.instance().handleHamburger();

    expect(mockHamburgerChange).toHaveBeenCalled();
  });

  it('should call loginWarning if not ready', () => {
    wrapper = shallow(<Hamburger 
      hamburgerChange={mockHamburgerChange}
      loginWarning={mockLoginWarning}
      ready={false}
    />);

    wrapper.instance().handleHamburger();

    expect(mockLoginWarning).toHaveBeenCalled();
  })
})