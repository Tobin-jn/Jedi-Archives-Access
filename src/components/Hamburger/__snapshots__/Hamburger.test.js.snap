// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger should match the snapshot 1`] = `
<section
  className="lightsaburger"
  onClick={[Function]}
>
  <div
    className="top "
  >
    <div
      className="top-blade blade "
    />
    <div
      className="top-grip grip "
    />
  </div>
  <div
    className="middle "
  >
    <div
      className="middle-grip grip "
    />
    <div
      className="middle-blade blade "
    />
  </div>
  <div
    className="bottom "
  >
    <div
      className="bottom-blade blade "
    />
    <div
      className="bottom-grip grip "
    />
  </div>
</section>
`;
